src/image-manipulation/effects.js

```diff
@@ -422,44 +430,70 @@ export function composite(src, x, y, cb) {
         return throwError.call(this, 'x and y must be numbers', cb);
     }
 
+    let { mode, opacitySource, opacityDest } = options;
```

Comment:
Can you add a test with these new params ?

Reply:
test added! it's only 1 can be more thorough if needed

Reply:
I would prefer to have a test that will expect the whole image to be equal to a base image so that we have more certainty that it will break fast if the code is wrong (like you've done in : https://github.com/oliver-moran/jimp/pull/524/files#diff-c632bfb61a4e129216f938dcb8d1a782R126)

Reply:
@edi9999 I added a better test but for some reason my bitmaps were always off by 1 before writing. The new test writes to the filesystem and checks those buffers.